### ISSUE TITLE
Standard usage pattern and registering users

### ISSUE THREAD

**Issue by ianpetzer** [levanto-financial/ember-intercom-io#5]
Hi,

Just curious whether standard usage for this would be to inject the {{ember-intercom-io}] component into the application.hbs if you want to use intercom throughout your app?

I'm asking because I load the registered user in asynchronously and the ember-intercom-io component has already initialised at this point, so I'm using the following code to stop and start the intercom service once the registered user details are available. Does this make sense, or am I missing something?

    loadCurrentUser: function() {
      if (this.get('session.isAuthenticated')) {      

        var currentUserId = this.get('session.session.content.authenticated.id');
        var user = this.store.findRecord('user', currentUserId).then(user => {

          this.get('intercom').set('user.name', user.get('name'));
          this.get('intercom').set('user.email', user.get('email'));
          this.get('intercom').set('user.createdAt', 1447135065173);              
          this.get('intercom').stop();
          this.get('intercom').start();
          this.set('session.currentUser', user);        
        })
      }
    }.observes('session.isAuthenticated')

**Comment by ianpetzer** [comment]
Following up on this.. the above code snippet allowed me to actually register a user on Intercom but stops the Intercom widget from appearing on my page

**Comment by psteininger** [comment]
@ianpetzer I am having the same issue on Ember 2.4.x. I think I will try another addon or simply do it manually

**Comment by mike-north** [comment]
Pull requests are welcome if you guys find problems :) Otherwise, I'll do my best to find time to make things easier for you, but more specificity would help (i.e., what are you looking to do, how have you attempted to implement it, what did you expect to happen, what actually happened).

**Comment by psteininger** [comment]
@mike-north @ianpetzer I think I figured out how to reliably set things up. 
First - I got rid of `{{intercom-io}}` - it might make sense to have that for unauthenticated users, but in my case we do not care (we use learn, and support, not acquire).

Then in `app/routes/application.js`:
```javascript
...
 intercom: Ember.inject.service(),
 actions: {
    didTransition(){
      if (this.get('session.isAuthenticated')){
        //handle intercom setup
        if (!this.get('intercom._hasUserContext')) {
          this.get('intercom').set('user.name', this.get('session.data.authenticated.profile.user_metadata.name'));
          this.get('intercom').set('user.email', this.get('session.data.authenticated.profile.email'));
          this.get('intercom').set('user.user_id', this.get('session.data.authenticated.profile.user_id'));
          this.get('intercom').set('user.createdAt', new Date(this.get('session.data.authenticated.profile.created_at')).getTime());
          this.get('intercom').start();
        } else {
          this.get('intercom').update();
        }
      }
    }
  }
...
```
We use `ember-simple-auth` for session service. This way is more reliable than using component. This could be easily adapted to extend to not-logged-in case, but intercom needs to be shut down and booted up again.

**Comment by jeffdill2** [comment]
@psteininger I see that you're setting `user.user_id` with your code. Is that actually working for you? When I try to set the `user_id` when creating the Intercom user, it doesn't come through successfully.

**Comment by seawatts** [comment]
You should be able to follow the new usage pattern in #30 
We still need to update the README to reflect this

**Comment by psteininger** [comment]
@jeffdill2  It turns out that it doesn't and it's because the code for the addon was written to ignore it. I had planned on making a change in a fork and do a PR. I got distracted with other things in our app, and still buried. I see that @seawatts already did us a favor. I will probably have time to update version next week and finally have user id.

**Comment by mike-north** [comment]
resolved in #30
